src/main/java/org/walkmod/javalang/compiler/types/IndexedURLClassPath.java

```diff
@@ -0,0 +1,152 @@
+package org.walkmod.javalang.compiler.types;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.*;
+import java.util.jar.JarEntry;
+import java.util.jar.JarFile;
+
+/**
+ * A modified URLClassPath that indexes the contents of classpath elements, for faster resource locating.
+ *
+ * The standard URLClassPath does a linear scan of the classpath for each resource, which becomes
+ * prohibitively expensive for classpaths with many elements.
+ */
+public class IndexedURLClassPath  {
+
+    private final URL[] urls;
+    private int lastIndexed = 0;
+    private static URL RT_JAR;
+    // Map from resource name to URLClassPath to delegate loading that resource to.
+    private final PathTree<URL> index = new PathTree<URL>();
+
+    static {
+        // static block to resolve java.lang package classes
+        String[] bootPath = System.getProperties().get("sun.boot.class.path").toString()
+                .split(Character.toString(File.pathSeparatorChar));
+        for (String lib : bootPath) {
+            if (lib.endsWith("rt.jar")) {
+                File f = new File(lib);
```

Comment:
Will that work for java 9?
Should we care now or later?

Reply:
Well, I would care if rt is not part of java 9

Reply:
The internal files rt.jar, tools.jar, and dt.jar have been removed. The content of these files is now stored in a more efficient format in implementation-private files in the lib directory. Class and resource files previously in tools.jar and dt.jar are now always visible via the bootstrap or application class loaders in a JDK image.

😱

Reply:
I am going to create a Issue, because it is a different discussion.

Reply:
https://github.com/rpau/javalang-compiler/issues/87